sequencer/node/src/node.rs

```diff
@@ -38,74 +31,30 @@ pub const CHANNEL_CAPACITY: usize = 1_000;
 const RPC_PORT_OFFSET: u16 = 1000;
 const ROUND_TIMEOUT_FOR_EMPTY_BLOCKS: u64 = 1500;
 
-struct CairoVMExecutionProgram {
-    // TODO: change this to a reference to a program and the casm contract class
-    // Vec<u8> is the bytes of the file of the cairo program
-    fib_program: Vec<u8>,
-    fact_program: Vec<u8>,
-}
-
-struct CairoNativeExecutionProgram {
-    fib_program: Arc<cairo_lang_sierra::program::Program>,
-    fact_program: Arc<cairo_lang_sierra::program::Program>,
-}
-
-impl CairoNativeExecutionProgram {
-    fn execute_fibonacci(&self, a: Vec<u32>, b: Vec<u32>, n: Vec<u32>) {
-        let ret = execute_fibonacci_cairo_native(&self.fib_program, a, b, n);
-        info!("Output Fib Cairo Native: ret is {:?}", ret)
-    }
-
-    fn execute_factorial(&self, n: Vec<u32>) {
-        let ret = execute_fact_cairo_native(&self.fact_program, n);
-        info!("Output Fact Cairo Native: ret is {:?}", ret)
-    }
-}
-
-impl CairoVMExecutionProgram {
-    fn execute_fibonacci(&self, n: usize) {
-        let ret = run_cairo_1_entrypoint(
-            self.fib_program.as_slice(),
-            0,
-            &[0_usize.into(), 1_usize.into(), n.into()],
-        );
-        info!("Output Fib CairoVM: ret is {:?}", ret)
-    }
-
-    fn execute_factorial(&self, n: usize) {
-        let ret = run_cairo_1_entrypoint(
-            self.fact_program.as_slice(),
-            0,
-            &[0_usize.into(), 1_usize.into(), n.into()],
-        );
-        info!("Output Fact CairoVM: ret is {:?}", ret)
-    }
-}
-
 enum ExecutionEngine {
-    Cairo(CairoVMExecutionProgram),
-    Sierra(CairoNativeExecutionProgram),
+    Cairo(Box<CairoVMEngine>),
+    Sierra(CairoNativeEngine),
 }
 
 impl ExecutionEngine {
-    fn execute_fibonacci(&self, a: usize, b: usize, n: usize) {
-        match self {
+    fn execute_fibonacci(&self, n: usize) {
+        let ret_msg = match self {
             ExecutionEngine::Cairo(execution_program) => execution_program.execute_fibonacci(n),
-            ExecutionEngine::Sierra(execution_program) => execution_program.execute_fibonacci(
-                get_input_value_cairo_native(a as u32),
-                get_input_value_cairo_native(b as u32),
-                get_input_value_cairo_native(n as u32),
-            ),
-        }
+            ExecutionEngine::Sierra(execution_program) => {
+                execution_program.execute_fibonacci(get_input_value_cairo_native(n))
```

Comment:
Just curious here:  Can't the API for ´execute_fib/fact´ be standardized to use the same types (ie. convert the type for CairoNative inside the execution engine). That way you won't need to match on the Engine type, maybe?

Reply:
We are addressing this in [#87](https://github.com/lambdaclass/starknet_stack/issues/87)